nixos/modules/services/hardware/nbfc-linux.nix

```diff
@@ -0,0 +1,61 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+
+with lib;
+
+let
+  cfg = config.services.nbfc-linux;
+
+  nbfcConfigList = splitString "\n" (
+    removeSuffix "\n" (
+      builtins.readFile (
+        pkgs.runCommand "nbfc-config-list" { } ''
+          ${pkgs.nbfc-linux}/bin/nbfc config --list > $out
+        ''
+      )
+    )
+  );
```

Comment:
This is IFD and not allowed in nixpkgs.

Reply:
Not sure what you mean by IFD. I'm doing this to validate that the provided `configName` is available in `nbfc-linux` (used in the assertions). Is there a better/nicer way of doing this?

Reply:
IFD, is import-from-derivation, which slows down evaluation and is disallowed inside nixpkgs.
You can give the configList option an enum type of all configs there are, but running the list command and reading the result is not allowed.